module/persistent.lua

```diff
@@ -0,0 +1,290 @@
+local awful = require("awful")
+local gobject = require("gears.object")
+local gtable = require("gears.table")
+local gtimer = require("gears.timer")
+local gfilesystem = require("gears.filesystem")
+local json = require(tostring(...):match(".*bling") .. ".helpers").json
+local tostring = tostring
+local string = string
+local ipairs = ipairs
+local pairs = pairs
+local table = table
+local capi = { awesome = awesome, root = root, screen = screen, client = client }
+
+local persistent = { }
+local instance = nil
+
+local function is_restart()
+    capi.awesome.register_xproperty("is_restart", "boolean")
+    local restart_detected = capi.awesome.get_xproperty("is_restart") ~= nil
+    capi.awesome.set_xproperty("is_restart", true)
+
+    return restart_detected
+end
+
+local function reapply_clients(self)
+    for index, client in ipairs(capi.client.get()) do
+        local pid =  tostring(client.pid)
+        if self.restored_settings.clients[pid] ~= nil then
+            -- Properties
+            local properties =
```

Comment:
Instead of having the same list twice in different functions and recreating that same "static" list on every function call, just move that list into the right scope so that you can access the same list in both places (here and `save_clients`)